src/lib/getCacher.ts

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+import { CacheData, Cacher } from '../types/Cacher'
+import { Options } from '../types/Options'
+import { print } from './logging'
+
+/**
+ * Check if cache is expired if timestamp is set
+ *
+ * @param cacheData
+ * @param cacheExpiration
+ * @returns
+ */
+function checkCacheExpiration(cacheData: CacheData, cacheExpiration = 10) {
+  if (typeof cacheData.timestamp !== 'number') {
+    return false
+  }
+
+  const unixMinuteMS = 60 * 1000
+  const expirationLimit = cacheData.timestamp + cacheExpiration * unixMinuteMS
+  return expirationLimit < Date.now()
+}
+
+export const defaultCacheFilename = '.ncu-cache.json'
+export const defaultCacheFile = `~/${defaultCacheFilename}`
+export const resolvedDefaultCacheFile = path.join(os.homedir(), defaultCacheFilename)
+
+/**
+ * The cacher stores key (name + version) - value (new version) pairs
+ * for quick updates across `ncu` calls.
+ *
+ * @returns
+ */
+export default function getCacher(options: Omit<Options, 'cacher'>): Cacher | undefined {
+  if (!options.cache) {
+    return
+  }
+
+  const file = options.cacheFile
```

Comment:
Let's use `options.cacheFile` inline since it is so short rather than introducing another variable.